### DISCUSSION TITLE
Rotate an element with martix?

### DISCUSSION THREAD

**Discussion by verberg** [cssbattle/forum#234] [General]
I'm just wondering, if it is possible to rotate an element with css transform:matrix(a,b,c,d,e,f).

**Comment by Beowolve** [comment]
You can rotate, skew,scale and translate, all in once with matrix.

A good tool to play with is here:
https://angrytools.com/css-generator/transform/

**Reply by verberg** [reply]
Oh, thanks, that's great!

**Comment by APerricone** [comment]
for rotate you can use [offset-path](https://developer.mozilla.org/en-US/docs/Web/CSS/offset), for example:

Rotate 90: `offset:path('M0,0v-1')`
Rotate 180: `offset:path('M0,0h-1')`
Rotate 270: `offset:path('M0,0v1')`
Rotate 45: `offset:path('M0,0l1 1')`
and you can use the M to positionate the element

on Firefox works also `rotate:90deg` [but it is not full supported](https://caniuse.com/mdn-css_properties_rotate)

**Reply by verberg** [reply]
Can you tell me, what's the offset-path for rotate -45 deg? 
And another question: Is this path the same as in SVG? Where to learn more about?

**Reply by rafehimself** [reply]
**`offset:path('mx,yl1-1')`**
where is x you move on horizontal axis with a number
where is y you move on vertical axis with a number

**Reply by verberg** [reply]
Oh! Thanks!

**Reply by midnightPush** [reply]
can someone explain or make tutorial about `clip-path:path( ...`
seems like it's used a lot here

**Reply by APerricone** [reply]
[MDN is the way](https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path).
there is also [Clippy](https://bennettfeely.com/clippy/)
